File: 프로그래머스/2/131704. 택배상자/택배상자.java
import java.util.*;
class Solution {
    public int solution(int[] order) {
        int answer = 0;
        int idx = 0;
        Stack<Integer> s = new Stack<>();
        for(int i = 1; i <= order.length; i++){
            s.push(i);
            while(!s.isEmpty()){
                if(s.peek() == order[idx]){
                    s.pop();
                    idx++;
                    answer++;
                }
                else{
                    break;
                }
            }
        }
        
        
        return answer;
    }
}

File: 프로그래머스/2/131704. 택배상자/택배상자.py
def solution(order):
    answer = 0
    belt = []
    idx = 0
    for box in range(1, len(order)+1):
        belt.append(box)
        while belt and (belt[-1]==order[idx]):
            answer+=1
            idx+=1
            belt.pop()
    return answer